lib/RequestHelper.php

```diff
@@ -13,52 +13,106 @@
 
 namespace OCA\Files_Antivirus;
 
+use OC\Cache\CappedMemoryCache;
+use OC\Files\Storage\Storage;
 use \OCP\IRequest;
 
+/**
+ * Used to detect the size of the uploaded file
+ *
+ * @package OCA\Files_Antivirus
+ */
 class RequestHelper {
-
 	/**
 	 * @var  IRequest
 	 */
 	private $request;
 
+	/**
+	 * @var CappedMemoryCache
+	 */
+	private static $fileSizeCache;
+
+	/**
+	 * RequestHelper constructor.
+	 *
+	 * @param IRequest $request
+	 */
 	public function __construct(IRequest $request) {
 		$this->request = $request;
 	}
 
+	/**
+	 * @return CappedMemoryCache
+	 */
+	public function getCache() {
+		if (self::$fileSizeCache === null) {
+			self::$fileSizeCache = new CappedMemoryCache();
+		}
+		return self::$fileSizeCache;
+	}
+
+	/**
+	 * @param string $path
+	 * @param string $size
+	 *
+	 * @return void
+	 */
+	public function setSizeForPath($path, $size) {
+		$this->getCache()->set($path, $size);
+	}
+
 	/**
 	 * Get current upload size
 	 * returns null for chunks and when there is no upload
 	 *
+	 * @param Storage $storage
 	 * @param string $path
 	 *
 	 * @return int|null
 	 */
-	public function getUploadSize($path) {
-		$uploadSize = null;
+	public function getUploadSize(Storage $storage, $path) {
+		// remove .ocTransferId444531916.part from part files
+		$cleanedPath = preg_replace(
+			'|\.ocTransferId\d+\.part$|',
+			'',
+			$path
+		);
+		// cache uses dav path in /files/$user/path format
+		$translatedPath = preg_replace(
+			'|^files/|',
+			'files/' . $storage->getOwner('/') . '/',
+			$cleanedPath
+		);
+		$cachedSize = $this->getCache()->get($translatedPath);
+		if ($cachedSize > 0) {
+			return $cachedSize;
+		}
 
 		$requestMethod = $this->request->getMethod();
+		// Are we uploading anything?
+		if ($requestMethod !== 'PUT') {
```

Comment:
maybe put this before all the regexp logic ?

Reply:
mhm. No, but I'd better wrap L75-90 with `if ($requestMethod !== 'MOVE') {`